src/bp-core/bp-core-attachments.php

```diff
@@ -1642,7 +1646,29 @@ function bp_attachments_cover_image_ajax_upload() {
 		);
 	}
 
-	$cover_url = trailingslashit( $bp_attachments_uploads_dir['baseurl'] ) . $cover_subdir . '/' . $cover['cover_basename'];
+	$cover_url = bp_attachments_get_attachment(
+		'url',
+		array(
+			'object_dir' => ( 'xprofile' === $component ) ? 'members' : $component,
+			'item_id'    => $item_id,
+		)
+	);
+
+	if ( '' === $cover_url ) {
+		$cover_url = trailingslashit( $bp_attachments_uploads_dir['baseurl'] ) . $cover_subdir . '/' . $cover['cover_basename'];
+	}
+
+	/**
+	 * Filters groups/members cover image attachment URL.
+	 *
+	 * @since BuddyBoss [BBVERSION]
+	 *
+	 * @param string $cover_url URL to the image.
+	 * @param array  $cover     Cover array.
+	 * @param string $component Component either groups or members.
+	 * @param int    $item_id   Inform about the item id the cover image was set for either group id or member id.
+	 */
+	$cover_url = apply_filters( 'bp_' . ( 'xprofile' === $component ) ? 'members' : $component . '_attachments_cover_image_url', $cover_url, $cover, ( 'xprofile' === $component ) ? 'members' : $component, $item_id );
```

Comment:
Looks like a broken ternary operation. The ternary expression that calculates the filter name will always just return 'member' and nothing else, it needs to be wrapped in paratheses. See this snippet: https://sandbox.onlinephpfunctions.com/c/87b8d

Reply:
@eriktorsner fixed now